src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -177,6 +177,9 @@ public virtual void WriteResponseHeaders([NotNull] OutputFormatterContext contex
                 throw new InvalidOperationException(Resources.FormatOutputFormatterNoMediaType(GetType().FullName));
             }
 
+            // Clone the media type as we don't want it to affect the next request
+            selectedMediaType = new MediaTypeHeaderValue(selectedMediaType.ToString());
```

Comment:
Would rather see this line of code **right** about the call to set CharSet. It would be easy to regress this by introducing some more logic here.